pkg/diagnostics/systemd/analyze_logs.go

```diff
@@ -0,0 +1,128 @@
+package systemd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+	"github.com/openshift/origin/pkg/diagnostics/types/diagnostic"
+)
+
+// AnalyzeLogs
+type AnalyzeLogs struct {
+	SystemdUnits map[string]types.SystemdUnit
+
+	Log *log.Logger
+}
+
+func (d AnalyzeLogs) Description() string {
+	return "Check for problems in systemd service logs since each service last started"
+}
+func (d AnalyzeLogs) CanRun() (bool, error) {
+	return true, nil
+}
+func (d AnalyzeLogs) Check() (bool, []log.Message, []error, []error) {
+	infos := []log.Message{}
+	warnings := []error{}
+	errors := []error{}
```

Comment:
@deads2k finally resurrecting this. I think I want to refactor the actual logging entirely out of the diagnostics themselves. This whole signature and intro are so ugly and awkward to interact with, and they don't even capture all the different log levels available or the order in which infos/errors/warnings are found.

Instead I think I want to return a *DiagnosticResult (new type). The role of the diagnostic would be to put messages, warnings, and errors into this result and set success. Then the thing running the diagnostic can decide what to actually log and when. This would help with parallelization too. That means you don't see partial logs produced during the run of an individual diagnostic, but I think I can live with that (and with parallel execution, that would have been semi-necessary anyway).

What do you think?